src/core_write.cpp

```diff
@@ -25,6 +26,16 @@ UniValue ValueFromAmount(const CAmount& amount)
             strprintf("%s%d.%08d", sign ? "-" : "", quotient, remainder));
 }
 
+UniValue ValueFromAmount(const arith_uint256& amount)
+{
+    bool sign = amount < 0;
```

Comment:
This might just be from my lack of insight into how arith_uint256 is implemented, but can amount ever be negative due to it representing a large unsigned value?

Reply:
Negative.

Reply:
I've now added a test case that tries to push in a very large number (used to render as negative), and then fixed the rendering by switching from use of `getdouble()`, to getting a `uint64` type.